onyo/lib/inventory.py

```diff
@@ -472,8 +490,9 @@ def modify_asset(self, asset: dict | UserDict | Path, new_asset: dict | UserDict
         # If a change in is.directory is implied, do this first:
         if asset.get("onyo.is.directory", False) != new_asset.get("onyo.is.directory", False):
             # remove or add dir aspect from/to asset
-            ops = self.add_directory(asset["onyo.path.absolute"]) if new_asset.get("onyo.is.directory", False)\
-                else self.remove_directory(asset["onyo.path.absolute"])
+            ops = self.add_directory(Item(asset["onyo.path.absolute"], repo=self.repo)) \
```

Comment:
Not sure, but I think `asset` should already be an `Item` at this point. Hence, `self.add_directory(asset)` should be fine. Did you check?

Reply:
See line 470: 
asset = self.get_asset(path) if isinstance(asset, Path) else asset

So, yes: `asset` already is the item you want to pass.

Reply:
It's not an `Item` though. It's a `UserDict`.

Reply:
Ok. Then leave it to be straightened in context of `set` (main user of `modify_asset`).